lib/src/config_provider/config.dart

```diff
@@ -215,363 +219,689 @@ class Config {
+                  _libraryImports,
+                );
+              },
+            )),
+        FixedMapKey(
+          key: strings.excludeAllByDefault,
+          valueSchema: BoolSchema(),
+          defaultValue: (node) => false,
+          resultOrDefault: (node) => _excludeAllByDefault = node.value as bool,
+        ),
+        FixedMapKey(
+          key: strings.sort,
+          valueSchema: BoolSchema(),
+          defaultValue: (node) => false,
+          resultOrDefault: (node) => _sort = node.value as bool,
+        ),
+        FixedMapKey(
+          key: strings.useSupportedTypedefs,
+          valueSchema: BoolSchema(),
+          defaultValue: (node) => true,
+          resultOrDefault: (node) => _useSupportedTypedefs = node.value as bool,
+        ),
+        FixedMapKey(
+          key: strings.comments,
+          valueSchema: _commentSchema(),
+          defaultValue: (node) => CommentType.def(),
+          resultOrDefault: (node) => _commentType = node.value as CommentType,
+        ),
+        FixedMapKey(
+          key: strings.name,
+          valueSchema: _dartClassNameStringSchema(),
+          defaultValue: (node) {
+            _logger.warning(
+                "Prefer adding Key '${node.pathString}' to your config.");
+            return 'NativeLibrary';
+          },
+          resultOrDefault: (node) => _wrapperName = node.value as String,
+        ),
+        FixedMapKey(
+          key: strings.description,
+          valueSchema: _nonEmptyStringSchema(),
+          defaultValue: (node) {
+            _logger.warning(
+                "Prefer adding Key '${node.pathString}' to your config.");
+            return null;
+          },
+          resultOrDefault: (node) => _wrapperDocComment = node.value as String?,
+        ),
+        FixedMapKey(
+            key: strings.preamble,
+            valueSchema: StringSchema(
+              result: (node) => _preamble = node.value as String?,
+            )),
+        FixedMapKey(
+          key: strings.useDartHandle,
+          valueSchema: BoolSchema(),
+          defaultValue: (node) => true,
+          resultOrDefault: (node) => _useDartHandle = node.value as bool,
+        ),
+        FixedMapKey(
+          key: strings.ffiNative,
+          valueSchema: OneOfSchema(
+            childSchemas: [
+              EnumSchema(allowedValues: {null}),
+              FixedMapSchema(
+                keys: [
+                  FixedMapKey(
+                    key: strings.ffiNativeAsset,
+                    valueSchema: StringSchema(),
+                    required: true,
+                  )
+                ],
+              )
+            ],
+            transform: (node) => ffiNativeExtractor(node.value),
+          ),
+          defaultValue: (node) => FfiNativeConfig(enabled: false),
+          resultOrDefault: (node) =>
+              _ffiNativeConfig = (node.value) as FfiNativeConfig,
+        ),
+      ],
+    );
+  }
```

Comment:
Do we have any if key `a` is in this map, then `b` is required?
Or if `a` is in this map, then `b` must _not_ be supplied?

That's probably not expressible in JSON schemas.

But I have some of that kind of logic in `package:native_assets_cli`. https://github.com/dart-lang/native/blob/main/pkgs/native_assets_cli/lib/src/model/build_config.dart

We would probably model that by referring from one schema to the other. I believe the setup in this PR would work fine with trying to add that.

Reply:
We can use `OneOfSchema` for these situations. It might be too complicated though for a lot of if-else, Since we'd need one Schema for each permutation.

Reply:
Another option supported by JsonSchema as well would be https://json-schema.org/understanding-json-schema/reference/conditionals.html#id6

Reply:
Ah nice! If I find some time, I can give your abstraction a spin and add these, and use it in package:native_assets_cli. If the abstraction works well standalone, we could move it to package:config_cli. 👌